lektor/build_programs.py

```diff
@@ -213,7 +215,7 @@ def iter_child_sources(self):
             if self.source.page_num is None:
                 child_sources.append(self._iter_paginated_children())
                 pq = p_config.get_pagination_query(self.source)
-                child_sources.append(set(all_children) - set(pq))
+                child_sources.append(x for x in all_children if x not in pq)
```

Comment:
This does something entirely different and will not be an O(1) lookup. Why was this change done?